cogs/displayinfo.py

```diff
@@ -33,85 +33,113 @@ class DisplayInfo(commands.Cog):
-                                print("task is now cancelled")
-                        self.timers[mem_before.name][self.current_date][mem_before.activity.name]["updateTask"] = None
-
-                        currentStartTime = self.timers[mem_before.name][self.current_date][mem_before.activity.name]["startTime"]
+            before_string_id = str(mem_before.id)
+            if before_string_id in self.timers:
+                if self.current_date in self.timers[before_string_id]:
+                    if mem_before.activity is not None and mem_before.activity.name in self.timers[before_string_id][self.current_date]:
+                        currentStartTime = self.timers[before_string_id][self.current_date][mem_before.activity.name]["startTime"]
+                        self.timers[before_string_id][self.current_date][mem_before.activity.name]["startTime"] = None
                         if currentStartTime:
                             playTime = datetime.now() - currentStartTime # Calculate the time difference (playTime) between the start and end time of the activity
-                            self.timers[mem_before.name][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[mem_before.name][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
+                            self.timers[before_string_id][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[before_string_id][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
                         print("end:", self.timers) # printing the dictionary to see the updates after an activity is ended
                         self.current_collection.replace_one({}, self.timers) # Update the MongoDB collection with the updated activity data(self.timers)
 
-    # This function is an asynchronous coroutine that runs indefinitely in the background when mem_after.activity is True.
-    # It is designed to update activity time information for a Discord member ('mem_after') whenever they start a new activity every 5 minutes.
-    async def timed_update(self, mem_before: discord.Member, mem_after: discord.Member):
-        await asyncio.sleep(300)
-        while not mem_before.activity and mem_after.activity:
-            playTime = 300 # add 5 mins to activity time
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["totalTime"] = self.timers[mem_after.name][self.current_date][mem_after.activity.name].get("totalTime", 0) + playTime
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["startTime"] = datetime.now()
-            print("in timed_update", self.timers) #printing the dictionary to see the timed updates
+    async def update_now(self, string_id):
```

Comment:
since we have if checks if the dictionary for the person is already created, it is probably best for this function to return True/False. True on success and False on no update was done.

Reply:
This can be done just by creating a variable. IE: `update_success = await self.update_now(string_id)`